src/utils/NetworkGuard.ts

```diff
@@ -38,28 +38,59 @@ const validate = (() => {
   };
 })();
 
-function isMCQOptions(obj: unknown): obj is MCQOptions {
+/**
+ * Validates the tags for a question
+ * Three types of tags:
+ * 1. Taxonomy: (i.e. Course: VETHUB2011)
+ * 2. Search tags: (i.e. Animal_Being) NO SPACES
+ * TODO: Decide on the format of the directives and its purpose
+ * 3. Directives: (i.e. !!EXCLUDE!!)
+ * NOTE: Tags w/ spaces will count as invalid. Use underscores instead. i.e. "SEM 1" should be "SEM_1"
+ * @param tag string to validate
+ * @returns {boolean}
+ */
+function isTag(tag: string): boolean {
+  const isTaxonomy = tag.includes(":") && tag.split(":").length === 2;
+  const isSearchTag = !tag.includes(":") && tag.trim().split(" ").length === 1;
+  return isTaxonomy || isSearchTag;
+}
+
+function isTags(arr: unknown): arr is DataTags {
```

Comment:
These two are being repeated except for 1 part:

Consider this:

```ts
function validateTags(arr: unknown, checkAll: boolean): boolean {
  if (!validate.isArray(arr, validate.isString)) {
    return false;
  }

  return checkAll ? arr.every(isTag) : arr.some(isTag);
}
```

Reply:
Also, please don't use unknown, otherwise what's the point in using TS?

Reply:
I used `unknown` here on purpose to validate its an array using the `is` keyword. (See `validate.isArray()`. 
There's a chance the `arr` variable from the api is not an array and so we validate that it's an array first before using `.every()` and `.some()`.
From ts docs: 
> unknown is the type-safe counterpart of any. Anything is assignable to unknown, but unknown isn’t assignable to anything but itself and any without a type assertion or a control flow based narrowing. Likewise, no operations are permitted on an unknown without first asserting or narrowing to a more specific type.

That's why I think its justified in this case.

[Some docs here on unknown and typeguards](https://docs.apify.com/academy/switching-to-typescript/unknown-and-type-assertions#:~:text=Why%20%22unknown%22%20is%20better%E2%80%8B,on%20values%20typed%20as%20unknown%20.)

Reply:
> These two are being repeated except for 1 part:
> 
> Consider this:
> 
> ```ts
> function validateTags(arr: unknown, checkAll: boolean): boolean {
>   if (!validate.isArray(arr, validate.isString)) {
>     return false;
>   }
> 
>   return checkAll ? arr.every(isTag) : arr.some(isTag);
> }
> ```

This is now implemented with the two other functions deleted.
Fixed.